### PR TITLE
Add ability to specify imagePullSecrets for Network Operator image

### PR DESCRIPTION
**Pull request by e0ne**
We need to imagePullSecrets while we use private or pulic image
registry which requires authorization.

Closes: #105

Signed-off-by: Ivan Kolodiazhnyi <ikolodiazhny@nvidia.com>

### CODE REVIEW

**Comment by adrianchiris** [comment]
No need for `imagePullSecrets` to propagate to the manifests ?

you may want to use a custom image for multus, device plugin, whereabouts etc

**Comment by e0ne** [comment]
@adrianchiris you're absolutely right. I was going to do it in a separate commit/PR, that's why I put 'Related issue' comment instead of 'Closes issue'

**Comment by adrianchiris** [comment]
> @adrianchiris you're absolutely right. I was going to do it in a separate commit/PR, that's why I put 'Related issue' comment instead of 'Closes issue'

ack, so better to cover it in one PR IMO (different commit) as its all related.

File: config/crd/bases/mellanox.com_nicclusterpolicies.yaml [hunk 1]
```diff
@@ -138,15 +150,19 @@ spec:
                 - version
                 type: object
               secondaryNetwork:
-                description: SecondaryNetwork describes configuration options for
-                  secondary network
+                description: SecondaryNetworkSpec describes configuration options
```
Comment (moshe010) [inline config/crd/bases/mellanox.com_nicclusterpolicies.yaml:153]: it called secondaryNetwork so why you change it to SecondaryNetworkSpec?  maybe we can just remove it and write "describes configuration options .."

Comment (e0ne) [reply config/crd/bases/mellanox.com_nicclusterpolicies.yaml:153]: Need to find a proper fix to make both kubebuilder and godoc happy

File: deployment/network-operator/README.md [hunk 2]
```diff
@@ -167,7 +169,8 @@ Production cluster environment can deny direct access to the Internet and instea
 | `nvPeerDriver.deploy` | bool | `false` | deploy NVIDIA Peer memory driver container |
 | `nvPeerDriver.repository` | string | `mellanox` | NVIDIA Peer memory driver image repository |
 | `nvPeerDriver.image` | string | `nv-peer-mem-driver` | NVIDIA Peer memory driver image name  |
-| `nvPeerDriver.version` | string | `1.0-9` | Mellanox OFED driver version  |
+| `nvPeerDriver.version` | string | `1.0-9` | NVIDIA Peer memory driver version  |
```
Comment (moshe010) [inline deployment/network-operator/README.md:172]: un releated to this PR but we update the nv peer mem to 1.1-0

**Review by moshe010 — APPROVED** [review]

**Pull request merged**